### ISSUE TITLE
Dark theme support for QCH output

### ISSUE THREAD

**Issue by BLumia** [doxygen/doxygen#9711]
**Describe the bug**
Not sure if it count as a bug, but the generated Qt QCH documentation doesn't work well inside Qt Creator 9.0

**Expected behavior**
The QCH documentation follows the Qt Creator's theme (light or dark), but not mixed, and it should work with the theme that doxygen provided by default instead of using a custom theme (since most distro won't provide a custom theme for documentation generation).

**Screenshots**
Currently it looks like:

![图片](https://user-images.githubusercontent.com/10095765/205600310-46f6a40d-db0f-4a4c-b097-856235f7228b.png)

**To Reproduce**
1. Install Qt Creator 9.0 or above, set Qt Creator to use dark theme.
2. Install a QCH documentation generated by doxygen with it's default theme.
3. See the documentation inside the IDE.

For convenience, most distro package KDE's libraries with documentation enabled, installing some KDE documentation under Arch Linux will be able to reproduce this issue.

**Version**
doxygen 1.9.5-1 from Arch Linux repo

**Additional context**
Somewhat related: https://github.com/doxygen/doxygen/issues/8404

But maybe we can have a better default theme that works in this case.

**Comment by BLumia** [comment]
![图片](https://user-images.githubusercontent.com/10095765/205679794-220d6596-23e4-42b7-9328-ae0be8daca84.png)

To make easier to test, I modified the test project used in my previous issue and ensure `HTML_COLORSTYLE=DARK` will be inside the Doxyfile that used to generate the doc. The above screenshot is the result. The following zip file is the project that can be used for testing this issue (You will also need Qt Creator 9 to load the QCH file and see the result).

[testdoxygen-dark.zip](https://github.com/doxygen/doxygen/files/10155615/testdoxygen-dark.zip)

> what setting could be used from the Qt Creator to pickup the color style?

Not sure, still not digging into it too much.

**Comment by albert-github** [comment]
It loos like there is no difference between what you see originally and what you now see when the DARK mode is set. 

The newest Qt  Creator that I have at my system is version 6.0.2) from the 6.2.4 Qt distribution.
I also have downloaded the 9.0.0 version (from GitHub) though this one doesn't start....

In the docs directory I can create the index.qhp file, but the question for me is how can I get the qhp file (or the html files together with all the js / css / inmage etc. files) into the QtCreator?
I can also create the qch file with the qhelpgenerator, but no idea how to load this either.

In the assistant from Qt everything looks OK.

**Comment by BLumia** [comment]
> do you know when the lighthtml was introduced ?

Not sure but I found a related commit: https://github.com/qt-creator/qt-creator/commit/5e73215a0af6a1bc74d3baebddce4ca20cd09579

**Comment by albert-github** [comment]
This is more about using litehtml as default, but definitely gives some potential clues to do further research.

**Comment by albert-github** [comment]
I looked at the code and see that QtWebEngine should be a possibility but probably isn't enabled during the build of QtCreator (as given in the "releases" om GitHub).

Another, alarming, issue is https://bugreports.qt.io/browse/QTCREATORBUG-24921?jql=project%20%3D%20QTCREATORBUG%20AND%20text%20~%20webengine%20ORDER%20BY%20created%20DESC%2C%20creator%20DESC as at the end it states:
> Qt Creator uses now litehtml for help module, so there is no need for webengine.

@doxygen do you know what is required to get doxygen output running with "litehtml" browsers?

**Comment by doxygen** [comment]
@albert-github From what I read [here](https://github.com/litehtml/litehtml) litehtml doesn't support many of the CSS style tags (the site links to a table). So I guess if we would want to support this renderer a whole new set of stylesheets has to be created to work around all the limitations. Maybe good to check if there is an older version of doxygen that still does work well enough, so we have at least a starting point.

**Comment by albert-github** [comment]
@doxygen The dark theme is of course new in 1.9.5 so no really old version available. The "light" 1.9.4 displays but has limitations how thing are show. I don't think that it will be possible at all to show the doxygen documentation of any (reasonable recent version) in the QtCreator.
I tried also the oldest release I could find on GitHub  of QtCreator (4.12.0) but here there is no webengine available either.

**Comment by miemoe** [comment]
No sure, if this is the right place to request more support for the QtCreator help generated output of doxygen.

I found this issue here by looking, if QtCreator Help renderer is somehow in focus for doxygen.
I would like to refer to an issue in Qt's Issue Tracker: https://bugreports.qt.io/browse/QTCREATORBUG-31672

The nice thing in QtCreator is, that if you position your cursor on a method (e.g. setX(...)) and press F1, then the right member is shown in the doxygen generated qt helper file.
In the referred issue a proposal was made to change the default css by adding:
`clear: left;`
to the
`.memitem`
to have the description boxes of the member functions placed to the left under its subject header.

**Comment by albert-github** [comment]
@miemoe 

I don't think it is right to add a request to an unrelated issue. The issue is for QtCreator but for something completely different.
So please create a new issue for it. Also
- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file!) that allows us to reproduce the problem? Please don't add external links as they might not be persistent (also references to GitHub repositories are considered non persistent).
- Please also specify the **full** doxygen version used (`doxygen -v`).


I don't think the recent versions of QtCreator are free to use so it will be a bit hard to test or am I mistaken and if so where to obtain an executable of a recent QtCreator version?
Same accounts for the qthelpgenerator /assistent and when free it uses a light HTML renderer that does not support all possibilities used by doxygen.
